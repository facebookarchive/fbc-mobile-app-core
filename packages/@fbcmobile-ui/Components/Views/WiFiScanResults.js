/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import * as React from 'react';
import Text from '@fbcmobile/ui/Components/Core/Text';
import WifiSignalIcon from '@fbcmobile/ui/Components/WifiSignalIcon';
import {Colors} from '@fbcmobile/ui/Theme';
import {Icon} from 'react-native-material-ui';
import {StyleSheet, View} from 'react-native';

export const WIFI_SECURITY_CAPABILITIES = ['WEP', 'WPA', 'WPA2', 'WPA_EAP'];

export type WiFiScanData = {|
  timestamp: number,
  frequency: number,
  channel: number,
  bssid: string,
  strength: number,
  ssid?: ?string,
  band?: ?string,
  channelWidth?: ?number,
  capabilities?: ?string,
  latitude?: ?number,
  longitude?: ?number,
  altitude?: ?number,
  heading?: ?number,
  rssi?: ?number,
|};

type Props = {|
  +scanResults: Array<WiFiScanData>,
|};

const WiFiScanResults = ({scanResults}: Props) => {
  return (
    <View style={styles.root}>
      {scanResults.map(wifi => {
        if (wifi != null) {
          const ssidText =
            wifi.ssid === '' || wifi.ssid == null ? '' : `${wifi.ssid}, `;
          return (
            <View style={styles.entry} key={wifi.bssid}>
              <Text color="gray" style={styles.ssidName}>
                {`${ssidText}${wifi.channel}-${wifi.band ?? ''}`}
              </Text>
              {wifi.capabilities != null &&
              WIFI_SECURITY_CAPABILITIES.some(s =>
                wifi.capabilities?.includes(s),
              ) ? (
                <Icon
                  style={styles.lockIcon}
                  name="lock"
                  iconSet="MaterialCommunityIcons"
                  color={Colors.Gray80}
                  size={16}
                />
              ) : null}
              <WifiSignalIcon signalStrength={wifi.strength} />
            </View>
          );
        }
      })}
    </View>
  );
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
  entry: {
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'flex-start',
    alignItems: 'center',
    paddingVertical: 12,
  },
  ssidName: {
    flexGrow: 1,
  },
  lockIcon: {
    marginRight: 4,
  },
});

export default WiFiScanResults;
