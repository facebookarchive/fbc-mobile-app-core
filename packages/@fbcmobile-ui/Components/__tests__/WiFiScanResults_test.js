/**
 * Copyright (c) Facebook, Inc. and its affiliates.
 * All rights reserved.
 *
 * This source code is licensed under the BSD-style license found in the
 * LICENSE file in the root directory of this source tree.
 *
 * @flow strict-local
 * @format
 */

'use strict';

import React from 'react';
import WiFiScanResults from '../Views/WiFiScanResults';
import renderer from 'react-test-renderer';
import type {WiFiScanData} from '../Views/WiFiScanResults';

describe('WiFiScanResults Tests', () => {
  test('empty CellScanResults renders correctly', () => {
    const tree = renderer.create(<WiFiScanResults scanResults={[]} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  test('CellScanResults renders correctly', () => {
    const results: WiFiScanData = {
      timestamp: 1234567890,
      frequency: 123,
      channel: 11,
      bssid: '11:22:33:44:55:66',
      strength: 10,
      ssid: 'My WiFi',
      capabilities: 'AES, WEP',
    };
    const tree = renderer
      .create(<WiFiScanResults scanResults={[results]} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
