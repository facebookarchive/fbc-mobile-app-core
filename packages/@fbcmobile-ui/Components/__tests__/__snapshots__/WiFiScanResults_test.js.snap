// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WiFiScanResults Tests CellScanResults renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "flexDirection": "row",
        "justifyContent": "flex-start",
        "paddingVertical": 12,
      }
    }
  >
    <Text
      style={
        Array [
          Array [
            Object {
              "color": "#484848",
              "fontSize": 16,
              "fontWeight": "500",
              "lineHeight": 20,
            },
            Object {
              "color": "#303846",
            },
            null,
            Object {
              "textAlign": "left",
            },
          ],
          Object {
            "flexGrow": 1,
          },
        ]
      }
    >
      My WiFi, 11-
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#484848",
            "fontSize": 16,
          },
          Object {
            "marginRight": 4,
          },
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰌾
    </Text>
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#484848",
            "fontSize": 16,
          },
          null,
          Object {
            "fontFamily": "Material Design Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      󰤨
    </Text>
  </View>
</View>
`;

exports[`WiFiScanResults Tests empty CellScanResults renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
